### ISSUE TITLE
String Manipulation

### ISSUE THREAD

**Issue by barnjamin** [algorand/pyteal-utils#1]
Some utility functions for string manipulation

`atoi`: convert "123" []byte into 123 uint64
`itoa`: convert 123 uint64 into "123" []byte
`strpos`: find location in string of substring, not sure what the right return value is if its not present? max(uint64)? 
`upper`: convert "abc" to "ABC"
`lower`: convert "ABC" to "abc"

Those are probably the most useful convenience functions to provide.

**Comment by barnjamin** [comment]
```python
@Subroutine(TealType.uint64)
def recursive_atoi(a):
    return If(
            Len(a) > Int(0),
            (
                ((GetByte(a, Int(0))) - Int(48) ) *
                Exp(Int(10), Len(a) - Int(1))
            ) + recursive_atoi( Substring(a,Int(1),Len(a)) ),
            Int(0)
        )
 ```

**Comment by barnjamin** [comment]
from gidon in discord

```python
@Subroutine(TealType.bytes)
def convert_uint_to_bytes(arg):

    string = ScratchVar(TealType.bytes)
    num = ScratchVar(TealType.uint64)
    digit = ScratchVar(TealType.uint64)

    return If(
        arg == Int(0),
        Bytes("0"),
        Seq([
            string.store(Bytes("")),
            For(num.store(arg), num.load() > Int(0), num.store(num.load() / Int(10))).Do(
                Seq([
                    digit.store(num.load() % Int(10)),
                    string.store(
                        Concat(
                            Substring(
                                Bytes('0123456789'),
                                digit.load(),
                                digit.load() + Int(1)
                            ),
                            string.load()
                        )
                    )
                ])

            ),
            string.load()
        ])
        ```

**Comment by barnjamin** [comment]
closed by #13

**Comment by barnjamin** [comment]
merged
